### ISSUE TITLE
parammeter can not be null errors

### ISSUE THREAD

**Issue by bennyok** [HewlettPackard/hpe-azurestack#1]
When running the script in a disconnected site with ADFS I get a lot of error with "$ratecardmapping.ContainsKey($azsmeters[$_.Properties.MeterId"  can not be null

**Comment by christophkarner** [comment]
The same situation in a connected scenario. Deployment tenant and billing tenant are different from each other. Stack is in a CSP multi-tenancy environment.
Maybe there are missing items in the $azsmeters hash table?

**Comment by christophkarner** [comment]
The errors seem to originate from missing meters in the hash table. I've made a pull request featuring the corresponding changes (added managed disk meters). The script now runs fine against my stamp.

**Comment by amcgregor1** [comment]
I believe the root cause is a logic error

If the expression ($azsmeters.ContainsKey($_.Properties.MeterId)) returns nothing (Null) then  

the condition ($ratecardmapping.ContainsKey($azsmeters[$_.Properties.MeterId])) will error as this will be $ratecardmapping.ContainsKey(Null))

I believe the logic should be, if ($azsmeters.ContainsKey($_.Properties.MeterId)) exists then test $ratecardmapping.ContainsKey($azsmeters[$_.Properties.MeterId]))
```PowerShell
if ($azsmeters.ContainsKey($_.Properties.MeterId)) {
       $record | Add-Member -Name MeterName -MemberType NoteProperty -Value $azsmeters[$_.Properties.MeterId]
       if($ratecardmapping.ContainsKey($azsmeters[$_.Properties.MeterId])) {
             $rateCardObj = $ratecardmapping[$azsmeters[$_.Properties.MeterId]]
             $record | Add-Member -Name RateCardMeterId -MemberType NoteProperty -Value $rateCardObj.MeterId
             $record | Add-Member -Name MeterUnit -MemberType NoteProperty -Value $rateCardObj.Units
             $record | Add-Member -Name OpenRate -MemberType NoteProperty -Value $rateCardObj.OpenRate
       } 
} else {
       $record | Add-Member -Name MeterName -MemberType NoteProperty -Value "Unknown"
       $record | Add-Member -Name RateCardMeterId -MemberType NoteProperty -Value "Unknown"
       $record | Add-Member -Name MeterUnit -MemberType NoteProperty -Value "Unknown"
       $record | Add-Member -Name OpenRate -MemberType NoteProperty -Value "0"
}
```

**Comment by vivekk2016** [comment]
Fixed the issue with commit cdbf7b09328d0a2e2e8eef4cbfe1c19681a7444f
